qualang_tools/digital_filters/filters.py

```diff
@@ -2,19 +2,30 @@
 from typing import Tuple, List
 import numpy as np
 import scipy.signal as sig
+from warnings import warn
+from enum import Enum
+
+
+class QOP_VERSION(Enum):
+    QOP222 = {
+        "feedforward_max": 2 - 2**-16,
+        "feedback_max": 1 - 2**-20,
+        "feedforward_length": lambda feedback_len: 44 - 7 * feedback_len,
+    }
 
 
 def calc_filter_taps(
     fir: List[float] = None,
     exponential: List[Tuple[float, float]] = None,
     highpass: List[float] = None,
     bounce: List[Tuple[float, float]] = None,
-    delay: int = None,
+    delay: float = None,
     Ts: float = 1,
+    qop_version: Enum = QOP_VERSION.QOP222,
```

Comment:
You're not passing these down to the other functions

Reply:
What do you mean?
Here I just call it at the end so that the final taps are checked.
I added it to the other functions definition in case people directly get the taps from them without calling `calc_filter_taps()`.

Reply:
Never I got what you meant and updated it :)

Reply:
I set it to NONE for the other function so that the check is only performed at the return